src/pool.rs

```diff
@@ -149,11 +230,37 @@ impl ConnectionPool {
         }
     }
 
-    // Connects to a MongoDB server as defined by the initial configuration.
-    fn connect(&self) -> Result<BufStream<TcpStream>> {
+
+    // Connects to a MongoDB server as defined by the initial
+    #[allow(unreachable_code)] // Suppresses warning for `panic` when ssl is enabled.
+    fn connect(&self) -> Result<BufStream<Stream>> {
         let host_name = &self.host.host_name;
         let port = self.host.port;
-        let stream = BufStream::new(try!(TcpStream::connect((&host_name[..], port))));
-        Ok(stream)
+        let inner_stream = try!(TcpStream::connect((&host_name[..], port)));
+
+        if let Some(SslConfig { ca_file: ref _ca_file,
+                                certificate_file: ref _certificate_file,
+                                key_file: ref _key_file }) = self.ssl {
+            #[cfg(feature = "ssl")]
+            {
+                let mut ssl_context = SslContext::builder(SslMethod::tls())?;
+                ssl_context.set_cipher_list("DEFAULT")?;
```

Comment:
any update on the cipher stuff here?

re: hostname verification, I've seen requests on things like reqwest and [rust-native-tls](https://github.com/sfackler/rust-native-tls/issues/13) to disable verification, but I agree that it's an option that can be added in the future if someone requests it. I believe it's set using [bitflags](https://github.com/sfackler/rust-openssl/pull/469/files#diff-a654b9fe1cba8cdeefbbd7d1e925b257R8).

```rust
ssl.param().set_hostflags(X509_CHECK_FLAG_NEVER_CHECK_SUBJECT);
```

re: ciphers, generally clients and servers both have their own accepted cipher list; if they have overlap, the connection is established and any of those overlapping ciphers can be chosen for encryption. These are defined as strings (see [the OpenSSL docs](https://www.openssl.org/docs/man1.0.2/apps/ciphers.html)), so you should be able to accept one in the SslConfig. If the user provides an invalid string, the openssl crate should error out accordingly (i'm guessing).